Logger/BaseLoggerMixins.cs

```diff
@@ -1,6 +1,61 @@
-﻿namespace Logger;
-
-public static class BaseLoggerMixins
-{
-
-}
+﻿namespace Logger;
+using System.Globalization;
+#nullable enable
+
+public static class BaseLoggerMixins
+{
+
+    public static void Error(this BaseLogger? logger, string message, params object[] arguments)
+    {
+        //check if null if so throw error
+        if(logger is null)
+        {
+            throw new System.ArgumentNullException(nameof(logger));
+        }
+        
+        //format string to return, use string.format so the argument is placed into the {0} spot
+        string formattedString = string.Format(CultureInfo.CurrentCulture, message, arguments);
+        
+        //Call log with specified loglevel for method and string
+        logger.Log(LogLevel.Error, formattedString);
+    }

```

Comment:
```suggestion
        if(logger is null)
        {
            throw new System.ArgumentNullException(nameof(logger));
        }
        
        //format string to return, use string.format so the argument is placed into the {0} spot
        string formattedString = string.Format(CultureInfo.CurrentCulture, message, arguments);
        
        //Call log with specified loglevel for method and string
        logger.Log(LogLevel.Error, formattedString);
    }

```
For these methods, instead checking for null and formatting each one, why not try to create a helper method that does all of these things to save you some code by calling the helper in each one instead.

Reply:
I initially meant to do that, but it looks like I forgot to implement it thanks for letting me know!